ast/src/analyzed/mod.rs

```diff
@@ -1151,6 +1155,7 @@ pub struct PolynomialReference {
 pub struct PolyID {
     pub id: u64,
     pub ptype: PolynomialType,
+    pub degree: Option<DegreeType>,
```

Comment:
Do we need the degree in the PolyID? This type should be as simple as possibe for performance reasons.

Reply:
Good to know, I'm was planning to do a second pass trying to remove stuff like this :)

Reply:
So I ended up introducing `RawPolyID` which has the current content of `PolyID` and is used everywhere in the execution data structures.

Reply:
Removed thanks